Fix radio other grouping

drusepth opened the pull request:
This PR includes the following changes:

- Binds our `other` radio boxes to the same model property as other radio selections on the field, ensuring the radio functionality of "if I click this one, it deselects all others" (c.f. https://github.com/dobtco/screendoor-v2/issues/4557)

- Adds an example radio field with a true `include_other_option` to our examples/index.html kitchen sink, for demonstrating the functionality of showing an "other" input when the other option is selected (and hidden again when a non-other option is selected):

![2017-07-18-201119_471x99_scrot](https://user-images.githubusercontent.com/538235/28332499-5ccc7990-6bf5-11e7-891f-0ec3b9352f63.png)

![2017-07-18-201134_374x98_scrot](https://user-images.githubusercontent.com/538235/28332503-612cda7a-6bf5-11e7-986c-a26ed0ba8934.png)

- Updates the tests to ensure the field serializes correctly when blank and when the other option is selected.

- Includes the latest dist/ build. 

It still needs to be tested from within Screendoor, but I think we can follow a similar update/test process to the approach we took in #142 (i.e. release new version upon PR merge, then test Screendoor on staging with it).

vlymar commented:
Looks like you decided not to update rivets?

drusepth commented:
Yeah, when I looked at upgrading Rivets it actually looked pretty straightforward and didn't seem to break anything, but since it wasn't required for this fix I figured we'd exclude it from the changes, get this fixed, and upgrade as a standalone change whenever we decided we wanted to.

drusepth commented:
@vlymar This is ready for review at your convenience.

vlymar commented on src/vendor_config.coffee:
```diff
@@ -18,6 +18,12 @@ rivets.binders.input =
 
+rivets.formatters.eq = (value, checkAgainst) ->
+  if value.constructor == Array
+    value[0] == checkAgainst
```
why does this only check the first value?

vlymar commented on src/templates/partials/options_field.eco:
```diff
@@ -10,10 +10,10 @@
   <div class='fr_option fr_other_option'>
     <label class='control'>
-      <input type='<%= fieldType %>' data-rv-checked='model.value.other_checked' />
+      <input type='<%= fieldType %>' data-rv-checkedarray="model.value.checked" value="<%= FormRenderer.t.other %>" />
```
can you explain the changes in this file?

drusepth commented:
@vlymar I updated the formatter to look at any value in the selected options, rather than just the first one, which lets the other textbox show up for checkbox fields that allow other as well (which was already working, but would have broken with this change -- good catch). I also added a test for checkboxes with an other option. 

Ready for you to review again.

This does mean we still have the minor bug where:

* If a user turns on the "Include an Other option" _and_ adds an option for "Other" manually,
* Then clicking either will select both and show the Other reason text box on the latter, e.g.

![2017-07-24-185754_716x373_scrot](https://user-images.githubusercontent.com/538235/28534761-1f27a2b6-70a2-11e7-88ef-4127d787eb2a.png)

I think that's a more minimal bug that we can tackle separately, if you think this fixes the more pressing bug.

vlymar commented:
LGTM 👍
